src/app/kbv/controllers/self-assessment-router.js

```diff
@@ -0,0 +1,42 @@
+const path = require("path");
+const BaseController = require("hmpo-form-wizard").Controller;
+const presenters = require("../../../presenters");
+
+class SelfAssessmentRouterController extends BaseController {
+  configure(req, res, next) {
+    req.form.options.template = path.join(
+      req.form.options.templatePath,
+      "self-assessment-router"
+    );
+    super.configure(req, res, next);
+  }
+
+  locals(req, res, callback) {
+    super.locals(req, res, (err, locals) => {
+      if (err) {
+        return callback(err, locals);
+      }
+
+      const translationKey = "self-assessment-router";
+      const items = Object.values(
+        req.translate(`fields.${translationKey}.items`)
+      );
+      const content = req.translate(`fields.${translationKey}.content`);
+
+      locals.question = {
```

Comment:
The idea of the present pattern is to abstract all of the complex logic into a separate utility class, which can be extensively tested. That's the approach that is shown in Experian KBVs - https://github.com/govuk-one-login/ipv-cri-kbv-front/blob/main/src/app/kbv/controllers/question.js#L32-L35

Also, by passing in req.translate into the presenter pattern the implementation can be abstracted away more cleanly.

Reply:
So the logic here was originally separated into its own presenter but as this is a one off special case lets say, the presenter wouldn't be reused anywhere else for what it needed to do. So it felt better to keep that logic in its controller. Typically when we are receiving a questionKey we directly load a question, however this scenario loads up an initial router page  which can go to two different pages still using the same questionKey. Presenters are currently set up to use the questionKey to grab the translations it needs. I could see two ways around it, amend all presenters to accept a manually passed in translationsKey or for this single scenario keep the logic in the controller. If you think it would be better to move this one off logic into a presenter thats not a problem?